poller-lambdas/src/index.ts

```diff
@@ -0,0 +1,86 @@
+import { SQSEvent } from 'aws-lambda';
+import { SendMessageCommand, SQSClient } from '@aws-sdk/client-sqs';
+import { POLLER_LAMBDA_ENV_VAR_KEYS, PollerId } from '../../shared/pollers';
+import { PollFunction } from './types';
+import EXAMPLE_long_polling from './pollers/EXAMPLE_long_polling';
+import EXAMPLE_fixed_frequency from './pollers/EXAMPLE_fixed_frequency';
+
+const getEnvironmentVariableOrCrash = (
+	key: keyof typeof POLLER_LAMBDA_ENV_VAR_KEYS,
+) => process.env[key]!;
+
+const sqs = new SQSClient({});
+const lambdaApp = process.env['App'];
+const ownQueueUrl = getEnvironmentVariableOrCrash(
+	POLLER_LAMBDA_ENV_VAR_KEYS.OWN_QUEUE_URL,
+);
+
+const queueNextInvocation = (props: {
+	MessageBody: string;
+	DelaySeconds?: number;
+}) =>
+	sqs.send(
+		new SendMessageCommand({
+			QueueUrl: ownQueueUrl,
+			MessageDeduplicationId: lambdaApp, // should prevent the same lambda from being invoked multiple times
+			...props,
+		}),
+	);
+
+const pollerWrapper =
+	(pollerFunction: PollFunction) =>
+	async ({ Records }: SQSEvent) => {
+		const startTimeEpochMillis = Date.now();
+		const secret = 'TODO'; //TODO get secret (using name from env var)
+		if (Records.length != 1) {
+			console.warn('Expected exactly one SQS record, but got', Records.length);
+		}
+		return Promise.all(
+			Records.map((record) => {
+				const valueFromPreviousPoll = record.body; //TODO maybe parse
+				return pollerFunction(secret, valueFromPreviousPoll)
+					.then(async (output) => {
+						const endTimeEpochMillis = Date.now();
+
+						await sqs.send(
+							new SendMessageCommand({
```

Comment:
this needs to accommodate adding multiple items to the queue, from one poller lambda payload

Reply:
done